### PR TITLE
Port forwarding guide refresh

### PR DESCRIPTION
**Pull request by vexr**
- Divided the previous "Required Ports" section into four distinct tables. This change aims to enhance the clarity and ease of navigation, helping users identify the correct ports to open based on whether they are running Space Acres, Advanced CLI, or specific roles like a Domain Operator or a Public RPC Server.

- Updated the content and layout, hopefully making it easier to follow and understand.

### CODE REVIEW

**Review by vexr — COMMENTED** [review]
If I can get some feedback on the "Required Ports for Autonomys Network" section for each component in the tables.  While I believe I've accurately captured the necessary information, I want to ensure everything is correct. 

Any insights would be greatly appreciated to confirm the accuracy and completeness of this section, as well as the rest of the guide.

File: docs/farming-&-staking/farming/additional-guides/network-info.md
Comment (EmilFattakhov) [inline docs/farming-&-staking/farming/additional-guides/network-info.md:62]:
```suggestion
| 9944   | TCP | Node (Consensus RPC) | - Exposes real-time and historical blockchain data. <br> - Enables transaction submissions through HTTP and WebSocket. Used for slot challenges. |
```

File: docs/farming-&-staking/farming/additional-guides/network-info.md
Comment (EmilFattakhov) [inline docs/farming-&-staking/farming/additional-guides/network-info.md:63]:
```suggestion
| 9945   | TCP | Node (Domain RPC) | - Exposes real-time and historical blockchain data. Enables transaction submissions through HTTP and WebSocket. Includes smart contracts.
```

File: docs/farming-&-staking/farming/additional-guides/network-info.md
Comment (EmilFattakhov) [inline docs/farming-&-staking/farming/additional-guides/network-info.md:26]: I believe we can omit this sentence, as it repeats the information from the last paragraph

File: docs/farming-&-staking/farming/additional-guides/network-info.md
Comment (EmilFattakhov) [inline docs/farming-&-staking/farming/additional-guides/network-info.md:43]: I believe we need to add port 9944 TCP, but @randywessels please double-check me on this

File: docs/farming-&-staking/farming/additional-guides/network-info.md
Comment (EmilFattakhov) [inline docs/farming-&-staking/farming/additional-guides/network-info.md:45]: I don't think it's correct - the domain node uses the archival node data, so port 30533 is required to get the archival history, I'm not sure if the domain operator node needs access to 30333, but it likely does. 

@randywessels once again, not something I really tested myself, something you likely know better than I do

**Review by EmilFattakhov — CHANGES_REQUESTED** [review]
Thanks @vexr, let's wait for the second review from @randywessels before jumping to any actions
But feel free to address suggestions (remove < br > in tables) to fix the broken preview

File: docs/farming-&-staking/farming/additional-guides/network-info.md
Comment (vexr) [reply docs/farming-&-staking/farming/additional-guides/network-info.md:39]: I based this off of a [message](https://discord.com/channels/864285291518361610/1062507270539321485/1211854604195598396) from Nazar that states port 9944 should not be forwarded.

Comment (EmilFattakhov) [reply docs/farming-&-staking/farming/additional-guides/network-info.md:39]: Agree, it should not be forwarded, however, is it used for node <> farmer communication. 
I guess it depends on whether we want to show all used ports, or the ports that needs to be forwarded only

Comment (vexr) [reply docs/farming-&-staking/farming/additional-guides/network-info.md:39]: Firewall instructions for port 9944 are currently covered in the `Linux Firewall Configuration` section, but it might be more appropriate to move this information into a dedicated `Firewall Rules` guide. Having firewall instructions in a Port Forwarding guide feels out of place. A separate guide would also allow us to include firewall rule instructions for Windows and other operating systems.

File: docs/farming-&-staking/farming/additional-guides/network-info.md
Comment (randywessels) [reply docs/farming-&-staking/farming/additional-guides/network-info.md:41]: This was a bit confusing for me as I had at one point several nodes running so I had to have unique ports for each.  I can confirm that 30333 and 30433 are needed for the node.  What I don't really know is about 30533.  I have it setup as a port trigger but I don't have a good way of seeing if it is being used.  We probably do need Nazar's input on that one.

As for 9944 and 9945, I am binding 9945 to my node container.  But it not forwarded.  And I have nothing for 9944.  But I think what I am really doing is using 9945 as how 9944 was intended if that makes sense.

I love the new format btw.  Much easier to understand.

File: docs/farming-&-staking/farming/additional-guides/network-info.md
Comment (randywessels) [reply docs/farming-&-staking/farming/additional-guides/network-info.md:39]: I think 9944 is only needed if you are running a node that is not on the same box as your farmer or you are using docker containers.  But yeah it wouldn't be forwarded, just opened on the firewall.

File: docs/farming-&-staking/farming/additional-guides/network-info.md
Comment (EmilFattakhov) [reply docs/farming-&-staking/farming/additional-guides/network-info.md]: @nazar-pc can you help us out when you can, please?

Comment (randywessels) [reply docs/farming-&-staking/farming/additional-guides/network-info.md]: @nazar-pc Do you mind taking a look at this?  It was mainly done to clean up the page and make it easier to understand but we also want to double check if all of these ports are actually required and we are providing accurate information.  Specifically 30533 and 30433 we had questions about.

[New Page](https://deploy-preview-642--subspace-docs-staging.netlify.app/docs/farming-&-staking/farming/additional-guides/networking/)

[Existing Page](https://docs.autonomys.xyz/docs/farming-&-staking/farming/additional-guides/networking)

Comment (nazar-pc) [reply docs/farming-&-staking/farming/additional-guides/network-info.md]: RPC description for farming an public RPC purposes should be different, currently it says this:
> Exposes real-time and historical blockchain data, enables transaction submissions through HTTP and WebSocket, and is used for slot challenges.

Part of which only makes sense for farming purposes (and even then I would not talk about slot challenges because that is not the only thing it is used for) and the other part only makes sense for public RPC and is completely irrelevant for farming purposes.

The only thing that is completely incorrect is P2P port of the domain, it is actually `30334`.

Comment (vexr) [reply docs/farming-&-staking/farming/additional-guides/network-info.md]: Changes proposed based on:
> RPC description for farming an public RPC purposes should be different, currently it says this:
> > Exposes real-time and historical blockchain data, enables transaction submissions through HTTP and WebSocket, and is used for slot challenges.

> Part of which only makes sense for farming purposes (and even then I would not talk about slot challenges because that is not the only thing it is used for) and the other part only makes sense for public RPC and is completely irrelevant for farming purposes.
- **Advanced CLI (Farmer)** Update the `Purpose` for 9944 to: `Enables transaction submissions through HTTP and WebSocket.`
- **RPC Server (Public)** Update the `Purpose` for 9944 to: `Allows retrieval of real-time and historical blockchain data.`

Changes proposed based on:
> The only thing that is completely incorrect is P2P port of the domain, it is actually 30334.
- **Advanced CLI (Domain Operator)** Remove `30433`, and add `30334`.

Would these changes be correct?

If I could also get some basic pointers as the the `Purpose` of the follow ports, I'd appreciate it.
- What would be a good updated `Purpose` be to describe port `9945` on a Public RPC (Domain) Server be? Something like `Allows retrieval of smart contract data.`?
- What would be a good `Purpose` to describe port `30334` on **Advanced CLI (Domain Operator)**?

Comment (nazar-pc) [reply docs/farming-&-staking/farming/additional-guides/network-info.md]: > * **Advanced CLI (Farmer)** Update the `Purpose` for 9944 to: `Enables transaction submissions through HTTP and WebSocket.`
> 
> * **RPC Server (Public)** Update the `Purpose` for 9944 to: `Allows retrieval of real-time and historical blockchain data.`

It is almost the opposite. For farmer it is sufficient to mention that it is needed for farmer operation, I don't see why more details are necessary. Farmer doesn't send any transactions there and doesn't use HTTP.

If node is used for RPC purposes then users will in fact send transactions to it and query all kinds of data from it (events, transactions, block contents, chain state, ...).

> **Advanced CLI (Domain Operator)** Remove `30433`, and add `30334`.

Don't touch `30433`. `40333` should be replaced with `30334`.

> What would be a good updated `Purpose` be to describe port `9945` on a Public RPC (Domain) Server be? Something like `Allows retrieval of smart contract data.`?

Similarly to node's RPC, but for domain: sending transactions (likely deploying or calling contracts, etc.). I don't really see the point of elaborating on it too much. Neither those who know what it is nor those who don't will benefit from it.

> What would be a good `Purpose` to describe port `30334` on **Advanced CLI (Domain Operator)**?

Just like for node, it is required for domain node to have good connectivity with P2P network.

---

Overall I would't elaborate on purpose too much. Here are more examples:

> Retrieve archival history from the blockchain.

Doesn't really retrieve history from the blockchain, it is used for both retrieving and serving pieces and metadata about archival history from P2P (farmers specifically).

> Synchronize with node peers to access archival history.

Technically farmers never reach out to nodes via P2P, nodes are only clients right now.

**Review by randywessels — APPROVED** [review]
After quite a bit of back and forth I think this looks great.  Thank you @vexr for all your hard work here.

**Pull request merged**
